src/components/gridContainer.js

```diff
@@ -456,7 +462,66 @@ class GridContainer extends Component {
             this.setState({statusMessage: message})
         }
     }
+
+    // validation methods to determine whether the table meets accessibility standards
+    tableHasAtLeastOneHeaderRowOrColumn() {
+        if (this.state.metaHeadercols == 0 && this.state.metaHeaderrows == 0) {
+            this.onError(no_headers_error_message);
+            return false;
+        }
+        return true;
+    }
     
+    tableHasAppropriateHeaderColumnsSet() {
+        const topRow = this.state.handsontableData[0];
+        const numberOfRequiredHeaderColumns = this.calculateNumberOfHeaderColumnsRequired(topRow)
+
+        if (parseInt(this.state.metaHeadercols) < numberOfRequiredHeaderColumns) {
+            this.onError(`Incorrect number of header columns set. Expected at least ${numberOfRequiredHeaderColumns}, got ${this.state.metaHeadercols}`);
+            return false;
+        }
+        return true
+    }
+
+    calculateNumberOfHeaderColumnsRequired(topRow) {
+        let required = 0;
+        
+        for (let i = 0; i < topRow.length; i++) {
+            if (!topRow[i]) {
+                required++;
+            } else {
+                break;
+            }
+        }
+        return required;
+    }
+
+    tableHasAppropriateHeaderRowsSet() {
+        let numberOfRequiredHeaderRows = this.calculateNumberOfHeaderRowsRequired(this.state.handsontableData);
+
+        if (parseInt(this.state.metaHeaderrows) < numberOfRequiredHeaderRows) {
+            this.onError(`Incorrect number of header rows set. Expected at least ${numberOfRequiredHeaderRows}, got ${this.state.metaHeaderrows}`);
+            return false;
+        }
+
+        return true
+    }
+
+    calculateNumberOfHeaderRowsRequired(data) {
```

Comment:
Hmm I'm not sure I understand this. The only way I think you could determine this number would be if there were also header columns and then checking for the empty cells diagonally from the top left until you find data?
If there are no header columns then it must be just at least one?